### PR TITLE
Switch codegen for kubernetesenv template off of bazel

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR begins the transformation of template-based codegen from bazel-based to go:generate based. The initial test case is the self-contained template used by the kubernetesenv adapter.

This PR:
- adds a template mode to mixer_codegen.sh
- adds a second go:generate line to kubernetesenv.sh to invoke template mode for the template file
- alters template name to match the adapter name (add 'env')
- updates the names of the generated files to be more friendly (less dictated by bazel)

If this is deemed acceptable, a follow-on PR will convert the remainder of the `mixer_proto_library` uses over to this model.

### CODE REVIEW

File: mixer/adapter/kubernetesenv/kubernetesenv.go [hunk 1]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 //go:generate $GOPATH/src/istio.io/istio/bin/mixer_codegen.sh -f mixer/adapter/kubernetesenv/config/config.proto
```
Comment (mandarjog) [inline mixer/adapter/kubernetesenv/kubernetesenv.go]: I wonder if 
```
//go:generate $GOPATH/src/istio.io/istio/bin/mixer_codegen.sh -a mixer/adapter/kubernetesenv
```
would be better? The script should check `-f config/config.proto` and `-t template/template.proto` ?

**Comment by mandarjog** [comment]
ok tested, other that my other comment, looks good.

**Comment by sdake** [comment]
LGTM after rebase.

File: mixer/adapter/kubernetesenv/kubernetesenv.go (hunk 1 above)
Comment (douglas-reid) [reply mixer/adapter/kubernetesenv/kubernetesenv.go]: I think having a convention for config + template for adapters is a decent idea. There are only two adapters currently in this situation currently in this codebase. If it is OK, let's put that in a follow-on PR.

**Pull request merged**